barretenberg/cpp/src/barretenberg/bb/main.cpp

```diff
@@ -165,6 +168,49 @@ template <IsUltraFlavor Flavor> bool proveAndVerifyHonk(const std::string& bytec
     return verifier.verify_proof(proof);
 }
 
+/**
+ * @brief Constructs and verifies a Honk proof for an acir-generated circuit
+ *
+ * @tparam Flavor
+ * @param bytecodePath Path to serialized acir circuit data
+ * @param witnessPath Path to serialized acir witness data
+ */
+template <IsUltraFlavor Flavor> bool proveAndVerifyHonk(const std::string& bytecodePath, const std::string& witnessPath)
+{
+    // Populate the acir constraint system and witness from gzipped data
+    auto constraint_system = get_constraint_system(bytecodePath);
+    auto witness = get_witness(witnessPath);
+
+    return proveAndVerifyHonkAcirFormat<Flavor>(constraint_system, witness);
+}
+
+/**
+ * @brief Constructs and verifies multiple Honk proofs for an ACIR-generated program.
+ *
+ * @tparam Flavor
+ * @param bytecodePath Path to serialized acir program data. An ACIR program contains a list of circuits.
+ * @param witnessPath Path to serialized acir witness stack data. This dictates the execution trace the backend should
+ * follow.
+ */
+template <IsUltraFlavor Flavor>
+bool proveAndVerifyHonkProgram(const std::string& bytecodePath, const std::string& witnessPath)
```

Comment:
Btw @ledwards2225 this is the one change I made in bberg for now. Just wanted to test that we could also prove the full stack that we generated during execution.